packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/JsiiRuntime.java

```diff
@@ -154,12 +155,18 @@ private JsonNode processCallbackResponse(final JsonNode resp) {
             } else {
                 error = e.getMessage();
             }
+
+            String errClass = e.getClass().toString();
+            name = errClass.equals("class software.amazon.jsii.JsiiError")
```

Comment:
Would recommend using `e.getClass().getCanonicalName()` here instead of just `toString`:

```suggestion
            String errClass = e.getClass().getCanonicalName();
            name = errClass.equals("software.amazon.jsii.JsiiError")
```

But actually, you could also just use an `instanceof` check:

```suggestion
            name = e instanceof JsiiError
```